pb_decode.c

```diff
@@ -1431,6 +1431,9 @@ static bool checkreturn pb_dec_bytes(pb_istream_t *stream, const pb_field_t *fie
 #ifndef PB_ENABLE_MALLOC
         PB_RETURN_ERROR(stream, "no malloc support");
 #else
+        if (stream->bytes_left < size)
+            PB_RETURN_ERROR(stream, "end-of-stream");
+
 #ifdef PB_ENABLE_ADV_SIZE_CHECK
         /* field.max_size_limit == 0 disables pre-mem alloc check */
         if (field->max_size_limit > 0 && alloc_size > field->max_size_limit){
```

Comment:
Since:
- getting here means `alloc_size > size`, 
- and `field->max_size_limit` is defined to a reasonable value on encoding side,
I don't understand why the `alloc_size > field->max_size_limit` test here didn't catch the too large value of `size` introduced by the fuzzer at decode time. Or is `field->max_size_limit` also randomized by the fuzzer at the same time as `size`?

Reply:
The fuzzer caught the first condition, but the backported patch introduced the two `if`, and I decided to backport the full patch instead of only one `if` and thinking about how to trigger the second one.

Reply:
I'm cool with the 2 fixes.